Library/Homebrew/cmd/audit.rb

```diff
@@ -548,12 +548,14 @@ def audit_github_repository
   end
 
   def audit_specs
-    if head_only?(formula) && formula.tap.to_s.downcase !~ /-head-only$/
-      problem "Head-only (no stable download)"
-    end
+    if formula.tap
```

Comment:
This is not what we want. WE want to filter head only tap, but check other tap or non tap formulae. Therefore I think we need to stick with 

```ruby
!(formula.tap && formula.tap.repo.downcase.end_with?("head-only"))
```